### PR TITLE
Refactor some book reading code

### PR DESCRIPTION
**Pull request by sacrificerXY**
#### Summary
SUMMARY: None

#### Purpose of change

I'm currently trying to make the player specific book reading code more generic so any `Character` can potentially use it.

#### Describe the solution

Move and refactor some book reading related functions to use `Character`.

#### Testing

These changes are covered by unit tests. They all passed. I also playtested a bit and everything works.

### CODE REVIEW

**Comment by wapcaplet** [comment]
As you saw, there are some unit tests around these. Would you be willing to rephrase those tests slightly, to put more emphasis on the newly refactored functions?

For example, since you have factored out `avatar::identify` from `do_read`, I suggest changing the "identifying unread books" test to call `identify` instead of `do_read`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/8d87f156b966b53a4dd5ae3e1bcc4dfbccb67685/tests/reading_test.cpp#L34-L36

Since `do_read` has so many other responsibilities, this yields a cleaner, more focused test.

And, since you have moved `fun_to_read` and `book_fun_for` into `Character`, it might be appropriate to change the "reading a book for fun" test to use a `Character` instance, instead of an `avatar` instance. That should be as simple as changing the `avatar dummy` line here:

https://github.com/CleverRaven/Cataclysm-DDA/blob/8d87f156b966b53a4dd5ae3e1bcc4dfbccb67685/tests/reading_test.cpp#L46-L48

to something like this:

```cpp
#include "player_helpers.h"     // defines clear_avatar()

clear_avatar();
Character &dummy = get_player_character();
```

File: src/avatar.cpp [hunk 1]
```diff
@@ -944,6 +889,79 @@ bool avatar::has_identified( const itype_id &item_id ) const
...
+                     skill.obj().name(), reading->req );
+        }
+    }
+
+    if( reading->intel != 0 ) {
+        add_msg( m_info, _( "Requires intelligence of %d to easily read." ), reading->intel );
+    }
+    //It feels wrong to use a pointer to *this, but I can't find any other player pointers in this method.
+    if( book_fun_for( book, *this ) != 0 ) {
+        add_msg( m_info, _( "Reading this book affects your morale by %d" ), book_fun_for( book, *this ) );
```
Comment (Night-Pryanik) [inline src/avatar.cpp:925]:
```suggestion
        add_msg( m_info, _( "Reading this book affects your morale by %d." ), book_fun_for( book, *this ) );
```

File: src/avatar.cpp [hunk 2]
```diff
@@ -944,6 +889,79 @@ bool avatar::has_identified( const itype_id &item_id ) const
...
+    }
+
+    if( book.type->use_methods.count( "MA_MANUAL" ) ) {
+        const matype_id style_to_learn = martial_art_learned_from( *book.type );
+        add_msg( m_info, _( "You can learn %s style from it." ), style_to_learn->name );
+        add_msg( m_info, _( "This fighting style is %s to learn." ),
+                 martialart_difficulty( style_to_learn ) );
+        add_msg( m_info, _( "It would be easier to master if you'd have skill expertise in %s." ),
+                 style_to_learn->primary_skill->name() );
+        add_msg( m_info, _( "A training session with this book takes %s" ),
```
Comment (Night-Pryanik) [inline src/avatar.cpp:935]:
```suggestion
        add_msg( m_info, _( "A training session with this book takes %s." ),
```

**Comment by sacrificerXY** [comment]
Okay. Will change the tests.

**Pull request merged**